src/app/ui/components/containers/modal/components/radix-dialog.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactNode } from 'react';
+
+import * as Dialog from '@radix-ui/react-dialog';
+import { css } from 'leather-styles/css';
+
+interface RadixDialogLayoutProps {
+  children: ReactNode;
+  title: ReactNode;
+  footer?: ReactNode;
+  onClose?(): void; // check as this always needs to be provided
+}
+export function RadixDialog({ children, onClose, title, footer }: RadixDialogLayoutProps) {
+  return (
+    <Dialog.Root open>
+      <Dialog.Portal>
+        <Dialog.Overlay
+          className={css({
+            backgroundColor: 'overlay',
+            position: 'fixed',
+            inset: 0,
+            animation: 'overlayShow 150ms cubic-bezier(0.16, 1, 0.3, 1)',
+          })}
+        >
+          <Dialog.Content
+            onPointerDownOutside={onClose}
+            className={css({
+              backgroundColor: 'accent.background-primary',
+              // remove borderRadius on small to give impression of full page
+              borderRadius: { base: '0', md: 'lg' },
+              boxShadow:
+                'hsl(206 22% 7% / 35%) 0px 10px 38px -10px, hsl(206 22% 7% / 20%) 0px 10px 20px -15px',
+              position: 'fixed',
+              top: '50%',
+              left: '50%',
+              transform: 'translate(-50%, -50%)',
+              width: { base: '100vw', md: '90vw' },
+              height: { base: '100vh', md: 'unset' },
+              maxWidth: { base: '100vw', md: '450px' },
+              maxHeight: { base: '100vh', md: '85vh' },
```

Comment:
I'm using the `panda` style of responsive objects as when I was using our hooks `useViewportMinWidth` I was noticing some extra re-renders and jumpy behaviour. 

For example on the `Recieve` modal, the UI was re-rendering and flashing when moving from `Receive` to `Receive BTC`